includes/payment-tokens/class-wc-stripe-payment-tokens.php

```diff
@@ -522,11 +522,10 @@ public function woocommerce_payment_token_set_default( $token_id ) {
 		$stripe_customer = new WC_Stripe_Customer( get_current_user_id() );
 
 		try {
-			if ( WC_Stripe_Feature_Flags::is_upe_checkout_enabled() ) {
-				if ( WC_Stripe_UPE_Payment_Gateway::ID === $token->get_gateway_id() ) {
-					$stripe_customer->set_default_payment_method( $token->get_token() );
-				}
-			} elseif ( WC_Gateway_Stripe::ID === $token->get_gateway_id() || WC_Gateway_Stripe_Sepa::ID === $token->get_gateway_id() ) {
+			// If the token is a payment method, set it as the default payment method. If the token is a source, set it as the default source.
+			if ( strpos( $token->get_token(), 'pm_' ) === 0 ) {
```

Comment:
Do you see us needing this logic anywhere else? If yes, it might be good to move these to helper functions, e.g. `is_source_token( $token )`, `is_payment_method_token( $token )`

Reply:
I didn't see any token check when I skimmed through the codebase. Keeping it as it is for now.